servicetalk-concurrent-jdkflow/src/main/java/io/servicetalk/concurrent/jdkflow/JdkFlowAdapters.java

```diff
@@ -0,0 +1,200 @@
+     * @param source {@link java.util.concurrent.Flow.Publisher} to convert to a {@link Publisher}.
+     * @param <T> Type of items emitted from the {@code source} and the returned {@link Publisher}.
+     * @return A {@link Publisher} representation of the passed {@link java.util.concurrent.Flow.Publisher}.
+     */
+    public static <T> Publisher<T> fromFlowPublisher(java.util.concurrent.Flow.Publisher<T> source) {
+        return new FlowPublisherToPublisher<>(source);
+    }
+
+    /**
+     * Converts the passed {@link Publisher} to a
+     * <a href="https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.html">JDK Flow</a>
+     * {@link java.util.concurrent.Flow.Publisher}.
+     *
+     * @param publisher {@link Publisher} to convert to a {@link Publisher}.
+     * @param <T> Type of items emitted from the {@code source} and the returned {@link Publisher}.
+     * @return A {@link Publisher} representation of the passed {@link PublisherSource}.
+     */
+    public static <T> java.util.concurrent.Flow.Publisher<T> toFlowPublisher(Publisher<T> publisher) {
+        return new PublisherToFlowPublisher<>(publisher);
+    }
+
+    /**
+     * Converts the passed {@link PublisherSource} to a
+     * <a href="https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.html">JDK Flow</a>
+     * {@link java.util.concurrent.Flow.Publisher}.
+     *
+     * @param source {@link PublisherSource} to convert to a {@link java.util.concurrent.Flow.Publisher}.
+     * @param <T> Type of items emitted from the {@code source} and the returned
+     * {@link java.util.concurrent.Flow.Publisher}.
+     * @return A {@link java.util.concurrent.Flow.Publisher} representation of the passed {@link PublisherSource}.
+     */
+    public static <T> java.util.concurrent.Flow.Publisher<T> toFlowPublisher(PublisherSource<T> source) {
+        return new PublisherSourceToFlowPublisher<>(source);
+    }
+
+    private static final class PublisherToFlowPublisher<T> implements java.util.concurrent.Flow.Publisher<T> {
+        private final PublisherSource<T> source;
+
+        PublisherToFlowPublisher(final Publisher<T> source) {
+            this.source = toSource(requireNonNull(source));
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class PublisherSourceToFlowPublisher<T> implements java.util.concurrent.Flow.Publisher<T> {
+        private final PublisherSource<T> source;
+
+        PublisherSourceToFlowPublisher(final PublisherSource<T> source) {
+            this.source = requireNonNull(source);
+        }
+
+        @Override
+        public void subscribe(final Subscriber<? super T> subscriber) {
+            source.subscribe(new FlowSubscriber<>(subscriber));
+        }
+    }
+
+    private static final class FlowSubscriber<T> implements PublisherSource.Subscriber<T> {
```

Comment:
nit: Rename to `StToFlowSubscriber` to match the other class names.

Reply:
This should then be FlowToStSubscriber